metabase/metabase
# fix pie chart labels are hidden when they should not be

Closes https://github.com/metabase/metabase/issues/48587

### Description

- Fix pie chart labels available space computation
- Truncate labels when they are too long

### How to verify

- Create a pie/sunburst chart with labels and percentages inside slices enabled
- Ensure it truncates labels instead of hiding them fully

### Demo

Top-left label is not visible even without truncation.

Before
<img width="518" alt="Screenshot 2024-10-21 at 9 49 58 PM" src="https://github.com/user-attachments/assets/c6183269-9de1-43f5-9c4a-8b50789d6b09">

After
<img width="535" alt="Screenshot 2024-10-21 at 9 49 34 PM" src="https://github.com/user-attachments/assets/587e86b3-3af5-42d5-97d2-0643e0352343">

Before
<img width="619" alt="Screenshot 2024-10-21 at 11 41 33 PM" src="https://github.com/user-attachments/assets/fdb38a40-af5b-48d4-b38e-c21f7839b3fb">

After
<img width="630" alt="Screenshot 2024-10-21 at 11 41 44 PM" src="https://github.com/user-attachments/assets/29c85b66-bc78-40cc-bf96-089d1a8b6807">

### Checklist
